src/LcdMenu.cpp

```diff
@@ -122,14 +134,18 @@ void LcdMenu::setBacklightBrightness(int level, bool persist)
 {
   _brightness = level;
 
-  //LOGV2(DEBUG_INFO, F("LCD: Writing %d as brightness"), _brightness  );
-  // analogWrite(10, _brightness);
-  //LOGV2(DEBUG_INFO, F("LCD: Wrote %d as brightness"), _brightness  );
+  #if DISPLAY_TYPE == DISPLAY_TYPE_LCD_KEYPAD && defined(LCD_BRIGHTNESS_PIN)
+    analogWrite(LCD_BRIGHTNESS_PIN, _brightness);
```

Comment:
Unfortunatelly analogWrite does not compile for me for the esp32. how did you build it? did you add some library which delegates analogWrite to LEDC?

Reply:
Well, DISPLAY_TYPE_LCD_KEYPAD also is not available with ESP32, so it probably wouldn't work anyway? analogWrite is a very basic Arduino function, it should exist on ESP32

Reply:
thats exactly what i thought as well ... but unfortunatelly analogWrite requires specific hardware capabilities which are apparently not supported by esp32: https://github.com/espressif/arduino-esp32/issues/4

i think i we leave `DISPLAY_TYPE_LCD_KEYPAD` unsupported for esp32, we should handle this with an `#error` message in configs. because this new code in LcdMenu.cpp is actually meant for esp + lcd keypad but it will not compile.

Reply:
Apologies, I missed this. I have an additional PR coming that includes configuration checking that disallows the ESP32/ DISPLAY_TYPE_LCD_KEYPAD configuration. An aside, but in the past I have used https://github.com/ERROPiX/ESP32_AnalogWrite to implement analogWrite on ESP32.